nixos/modules/services/networking/ssh/ssh3-server.nix

```diff
@@ -0,0 +1,99 @@
+{ pkgs, config, lib, ... }:
+let
+  inherit (lib)
+    mkEnableOption
+    mkIf
+    mkOption
+    mkPackageOptionMD
+    types;
+  cfg = config.services.ssh3-server;
+  acmeDirectory = config.security.acme.certs.${cfg.domain}.directory;
+in
+  {
+    options.services.ssh3-server = {
+      enable = mkEnableOption "ssh3-server, a SSH3 (QUIC+TLS 1.3) server for secure remote logins";
+
+      package = mkPackageOptionMD pkgs "ssh3" { };
+      listenAddresses = mkOption {
+        type = with types; listOf (submodule {
+          options = {
+            addr = mkOption {
+              type = str;
+              description = ''
+                Host, IPv4 or IPv6 address to listen to.
+              '';
+            };
+            port = mkOption {
+              type = port;
+              description = ''
+                Port to listen to.
+              '';
+            };
+          };
```

Comment:
Actually, I don't see `addr` or `port` used individually, it might be better to just make it a list of strings of the form `addr:port`

Reply:
This means that you need to reparse the data if you need to introspect config for some reason. I would find this a bit bizarre, is it a nitpick or a blocker in your eyes? :P

Reply:
Not a blocker, just a note

Reply:
Resolving for now, I don't think it's very important to solve